apps/api/src/domain/models/audio-visual/shared/queries/transcript-query-builder.ts

```diff
@@ -0,0 +1,243 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { MultilingualTextItemRole, ResourceType } from '@coscrad/api-interfaces';
+import { AqlQuery } from 'arangojs/aql';
+import { buildMultilingualTextWithSingleItem } from '../../../../../domain/common/build-multilingual-text-with-single-item';
+import { MultilingualTextItem } from '../../../../../domain/common/entities/multilingual-text';
+import { AggregateId } from '../../../../../domain/types/AggregateId';
+import { InternalError } from '../../../../../lib/errors/InternalError';
+import { AudiovisualResourceType } from '../../audio-item/entities/audio-item-composite-identifier';
+import { TranslationLineItemDto } from '../../audio-item/queries/audio-item-query-repository.interface';
+import { TranscriptLineItemDto, TranslationItem } from '../commands';
+import { TranscriptItem } from '../entities/transcript-item.entity';
+import { TranscriptParticipant } from '../entities/transcript-participant';
+import { Transcript } from '../entities/transcript.entity';
+
+export class ArangoTranscriptQueryBuilder {
+    private readonly collectionName: string;
+
+    constructor(resourceType: AudiovisualResourceType) {
+        if (resourceType === ResourceType.audioItem) {
```

Comment:
This logic definitely doesn't belong here. Let's be sure not to duplicate it anywhere else before deciding how to handle this. It ties in with dynamic registration of view types \ collections.